libs/components/forms/src/lib/modules/checkbox/checkbox-group.component.ts

```diff
@@ -69,22 +69,26 @@ export class SkyCheckboxGroupComponent {
   public headingHidden = false;
 
   /**
-   * The semantic heading level in the document structure.
+   * The semantic heading level in the document structure. By default, the heading text is not wrapped in a heading element.
    * @preview
-   * @default 3
    */
   @Input({ transform: numberAttribute })
-  public headingLevel: SkyCheckboxGroupHeadingLevel = 3;
+  public set headingLevel(value: SkyCheckboxGroupHeadingLevel | undefined) {
+    this.#_headingLevel = value && !isNaN(value) ? value : undefined;
```

Comment:
Why is the `isNaN` check needed? The transform will mean that it is always either a number or undefined, correct?

Reply:
no, unfortunately the transform means it is either a number or `NaN` 😭 so even when the consumer doesn't set it and you would think it was undefined, it is actually `NaN`. `undefined` is easier to work with everywhere else so I transform it in the setter

Reply:
Ah. Ok. Makes sense.